### ISSUE TITLE
Accuracy changes with number of GPUs

### ISSUE THREAD

**Issue by abhinavs95** [apache/incubator-mxnet#11919]
## Description
I trained same SqueezeNet model with same hyper-parameters and dataset on p3.8xlarge and p3.16xlarge with same AMI but got ~3% lower accuracies on p3.16xlarge. I used same batch size per GPU but effective batch size is 2x in p3.16xlarge due to 2x number of GPUs.

## Environment info (Required)

p3.8xlarge
```
----------Python Info----------
Version      : 3.6.6
Compiler     : GCC 7.2.0
Build        : ('default', 'Jun 28 2018 17:14:51')
Arch         : ('64bit', '')
------------Pip Info-----------
Version      : 10.0.1
Directory    : /home/ubuntu/anaconda3/envs/gln/lib/python3.6/site-packages/pip
----------MXNet Info-----------
Version      : 1.3.0
Directory    : /home/ubuntu/anaconda3/envs/gln/lib/python3.6/site-packages/mxnet
Commit Hash   : 65fee984437dcca3516912417e9430cf34ba7313
----------System Info----------
Platform     : Linux-4.4.0-1062-aws-x86_64-with-debian-stretch-sid
system       : Linux
node         : ip-172-31-78-153
release      : 4.4.0-1062-aws
version      : #71-Ubuntu SMP Fri Jun 15 10:07:39 UTC 2018
----------Hardware Info----------
machine      : x86_64
processor    : x86_64
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                32
On-line CPU(s) list:   0-31
Thread(s) per core:    2
Core(s) per socket:    16
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 79
Model name:            Intel(R) Xeon(R) CPU E5-2686 v4 @ 2.30GHz
Stepping:              1
CPU MHz:               1972.070
CPU max MHz:           3000.0000
CPU min MHz:           1200.0000
BogoMIPS:              4600.11
Hypervisor vendor:     Xen
Virtualization type:   full
L1d cache:             32K
L1i cache:             32K
L2 cache:              256K
L3 cache:              46080K
NUMA node0 CPU(s):     0-31
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single kaiser fsgsbase bmi1 hle avx2 smep bmi2 erms invpcid rtm rdseed adx xsaveopt
----------Network Test----------
Setting timeout: 10
Timing for MXNet: https://github.com/apache/incubator-mxnet, DNS: 0.0032 sec, LOAD: 0.3394 sec.
Timing for Gluon Tutorial(en): http://gluon.mxnet.io, DNS: 0.1960 sec, LOAD: 0.3322 sec.
Timing for Gluon Tutorial(cn): https://zh.gluon.ai, DNS: 0.1599 sec, LOAD: 0.5460 sec.
Timing for FashionMNIST: https://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/gluon/dataset/fashion-mnist/train-labels-idx1-ubyte.gz, DNS: 0.0474 sec, LOAD: 0.7632 sec.
Timing for PYPI: https://pypi.python.org/pypi/pip, DNS: 0.0039 sec, LOAD: 0.1079 sec.
Timing for Conda: https://repo.continuum.io/pkgs/free/, DNS: 0.0043 sec, LOAD: 0.0531 sec.
```

p3.16xlarge

```
----------Python Info----------
Version      : 3.6.6
Compiler     : GCC 7.2.0
Build        : ('default', 'Jun 28 2018 17:14:51')
Arch         : ('64bit', '')
------------Pip Info-----------
Version      : 10.0.1
Directory    : /home/ubuntu/anaconda3/envs/gluon/lib/python3.6/site-packages/pip
----------MXNet Info-----------
Version      : 1.3.0
Directory    : /home/ubuntu/anaconda3/envs/gluon/lib/python3.6/site-packages/mxnet
Commit Hash   : 3051c49e3454df3b5f8909d3d76c6213d13539ad
----------System Info----------
Platform     : Linux-4.4.0-1062-aws-x86_64-with-debian-stretch-sid
system       : Linux
node         : ip-172-31-45-182
release      : 4.4.0-1062-aws
version      : #71-Ubuntu SMP Fri Jun 15 10:07:39 UTC 2018
----------Hardware Info----------
machine      : x86_64
processor    : x86_64
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                64
On-line CPU(s) list:   0-63
Thread(s) per core:    2
Core(s) per socket:    16
Socket(s):             2
NUMA node(s):          2
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 79
Model name:            Intel(R) Xeon(R) CPU E5-2686 v4 @ 2.30GHz
Stepping:              1
CPU MHz:               1581.609
CPU max MHz:           3000.0000
CPU min MHz:           1200.0000
BogoMIPS:              4600.07
Hypervisor vendor:     Xen
Virtualization type:   full
L1d cache:             32K
L1i cache:             32K
L2 cache:              256K
L3 cache:              46080K
NUMA node0 CPU(s):     0-15,32-47
NUMA node1 CPU(s):     16-31,48-63
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq monitor est ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single kaiser fsgsbase bmi1 hle avx2 smep bmi2 erms invpcid rtm rdseed adx xsaveopt ida
----------Network Test----------
Setting timeout: 10
Timing for MXNet: https://github.com/apache/incubator-mxnet, DNS: 0.0032 sec, LOAD: 0.3625 sec.
Timing for Gluon Tutorial(en): http://gluon.mxnet.io, DNS: 0.1710 sec, LOAD: 0.3294 sec.
Timing for Gluon Tutorial(cn): https://zh.gluon.ai, DNS: 0.0788 sec, LOAD: 0.5925 sec.
Timing for FashionMNIST: https://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/gluon/dataset/fashion-mnist/train-labels-idx1-ubyte.gz, DNS: 0.0093 sec, LOAD: 0.3598 sec.
Timing for PYPI: https://pypi.python.org/pypi/pip, DNS: 0.0035 sec, LOAD: 0.0962 sec.
Timing for Conda: https://repo.continuum.io/pkgs/free/, DNS: 0.0036 sec, LOAD: 0.0431 sec.
```
Package used: Python

## Minimum reproducible example
Used model definition and training script from GluonCV - https://gluon-cv.mxnet.io/build/examples_classification/dive_deep_imagenet.html#sphx-glr-build-examples-classification-dive-deep-imagenet-py

## Steps to reproduce

1. p3.8xlarge - `python train_imagenet.py --data-dir ../imagenet/img_dataset --batch-size 128 --num-gpus 4 -j 32 --mode hybrid --num-epochs 100 --lr 0.01 --wd 0.0002 --lr-decay 0.1 --lr-decay-epoch 60,90 --model squeezenet1.0`

2. p3.16xlarge - `python train_imagenet.py --data-dir ../imagenet/img_dataset --batch-size 128 --num-gpus 8 -j 64 --mode hybrid --num-epochs 100 --lr 0.01 --wd 0.0002 --lr-decay 0.1 --lr-decay-epoch 60,90 --model squeezenet1.0`

**Comment by szha** [comment]
Since the actual batch size differs by 2x it's not surprising that accuracy can be different.

**Comment by rahul003** [comment]
You would need to change your learning rate based on the total batch size, generally proportional to the batch size (as the number of steps the training takes halves in the latter case). Try using lr 0.02 for the latter case.

**Comment by Roshrini** [comment]
@abhinavs95 Did you try the above suggestions? 
@sandeep-krishnamurthy Can you please add label: Performance, Gluon
